samza-core/src/main/java/org/apache/samza/storage/TaskSideInputHandler.java

```diff
@@ -259,6 +277,44 @@ public void stop() {
     return oldestOffsets;
   }
 
+  /**
+   * Checks if whether the given offset for the SSP has reached the latest offset (determined at init),
```

Comment:
please clarify what caught up means - because when starting offset = newest offset - it is caught up but the comparator result says not caught up. Oe does consuming newset offset qualify as caught up?

Reply:
+1 I will clarify the doc.